### ISSUE TITLE
[1.16.5] Save Map button creates NumberFormatException

### ISSUE THREAD

**Issue by ejm554** [TeamJM/journeymap#410]
#### JourneyMap Version:

journeymap-1.16.5-5.7.3

#### Forge Version:

forge-1.16.5-36.2.20

#### Describe the problem in repeatable steps:

1. In Minecraft client running JourneyMap, go to Fullscreen map.
2. Click 'Save Map' button.
3. Observe NumberFormatException error message in log file.
4. Notice also that an image file is not saved.

#### Relevant errors from `.minecraft/journeymap/journeymap.log`:
```
[11:24:09] [Render thread/ERROR] [MapSaver] java.lang.NumberFormatException: For input string: "-1 - Copy"
	at java.base/java.lang.NumberFormatException.forInputString(Unknown Source)
	at java.base/java.lang.Integer.parseInt(Unknown Source)
	at java.base/java.lang.Integer.parseInt(Unknown Source)
	at journeymap.client.io.MapSaver.prepareFiles(MapSaver.java:198)
	at journeymap.client.io.MapSaver.<init>(MapSaver.java:75)
	at journeymap.client.ui.fullscreen.Fullscreen.lambda$initButtons$25(Fullscreen.java:723)
	at net.minecraft.client.gui.widget.button.Button.func_230930_b_(SourceFile:33)
	at journeymap.client.ui.component.Button.func_230930_b_(Button.java:382)
	at net.minecraft.client.gui.widget.button.AbstractButton.func_230982_a_(SourceFile:16)
	at net.minecraft.client.gui.widget.Widget.func_231044_a_(Widget.java:136)
	at journeymap.client.ui.component.Button.func_231044_a_(Button.java:393)
	at net.minecraft.client.gui.INestedGuiEventHandler.func_231044_a_(SourceFile:27)
	at journeymap.client.ui.fullscreen.Fullscreen.func_231044_a_(Fullscreen.java:1002)
	at net.minecraft.client.MouseHelper.func_198033_b(MouseHelper.java:87)
	at net.minecraft.client.gui.screen.Screen.func_231153_a_(Screen.java:427)
	at net.minecraft.client.MouseHelper.func_198023_a(MouseHelper.java:85)
	at net.minecraft.client.MouseHelper.func_228030_c_(MouseHelper.java:181)
	at net.minecraft.util.concurrent.ThreadTaskExecutor.execute(SourceFile:94)
	at net.minecraft.client.MouseHelper.func_228028_b_(MouseHelper.java:180)
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:36)
	at org.lwjgl.system.JNI.invokeV(Native Method)
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3101)
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(SourceFile:102)
	at net.minecraft.client.MainWindow.func_227802_e_(MainWindow.java:305)
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:997)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:607)
	at net.minecraft.client.main.Main.main(Main.java:184)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:51)
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37)
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54)
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72)
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82)
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at io.github.zekerzhayard.forgewrapper.installer.Main.main(Main.java:57)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:210)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:245)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)
```

**Comment by mysticdrew** [comment]
Can you see if this is an issue in 5.8.0beta18?

**Comment by ejm554** [comment]
Yes, I can check. Do I need to install the beta on both my server *and* my client? Or can I run it on just one side?

**Comment by mysticdrew** [comment]
Both, or check singleplayer

**Comment by mysticdrew** [comment]
I just checked and it is working for me.

**Comment by ejm554** [comment]
> I just checked and it is working for me...

Do you mean it's working for you in the release version?

**Comment by mysticdrew** [comment]
> > I just checked and it is working for me...
> 
> Do you mean it's working for you in the release version?

For me in my development version(beta18).

**Comment by ejm554** [comment]
I ran it again on our server, using the same world, and this happened:
- Following installation "out of the box," I was able to save an image of the map.
- After copying my JourneyMap files from the previous version to the new JourneyMap directory, the same error occurred.

**Comment by ejm554** [comment]
I think I may have found the problem. I searched the JourneyMap directory for the string "-1 - Copy" as noted in the log, and I found an image file where that was part of its filename. No idea where it came from, but after deleting the file, the Save Map button worked as intended. This was performed in the Release version, BTW, not the Beta. 

Strangely, the date attribute had today's date on it, but it was set to a time hours before I was even active on my device. I'm not sure if this indicates that the file was automatically generated somehow or if it a result of human error.

![image](https://user-images.githubusercontent.com/26723535/158076272-65cb1af9-8743-41a1-9edd-f030ae62b71a.png)
